packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/edge/EdgeLabel.tsx

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { EdgeLabelProps } from './EdgeLabeL.types';
+
+export const EdgeLabel = ({ transform, label, edgeLabelStyle }: EdgeLabelProps) => {
+  const style: React.CSSProperties = {
+    position: 'absolute',
+    background: 'transparent',
+    padding: 10,
+    transform,
+    zIndex: 999,
+  };
+
+  const { bold, italic, strikeThrough, underline, fontSize, color } = edgeLabelStyle;
+  if (bold) {
+    style.fontWeight = 'bold';
+  }
+  if (italic) {
+    style.fontStyle = 'italic';
+  }
+
+  if (fontSize) {
+    style.fontSize = fontSize;
+  }
+  if (color) {
+    style.color = color;
+  }
+
+  let decoration: string = '';
+  if (strikeThrough) {
+    decoration = decoration + 'line-through';
+  }
+  if (underline) {
+    const separator: string = decoration.length > 0 ? ' ' : '';
+    decoration = decoration + separator + 'underline';
+  }
+  if (decoration.length > 0) {
+    style.textDecoration = decoration;
+  }
```

Comment:
In a similar fashion, I did not notice but this kind of code is done in convertDiagram, not in the component. Which is why we don't have the GQL types here.

Reply:
Remove from here